crowbar_framework/spec/lib/runtime_checks_spec.rb

```diff
@@ -0,0 +1,108 @@
+#
+# Copyright 2015, SUSE LINUX Products GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "spec_helper"
+require "logger"
+require "byebug"
+
+describe RuntimeChecks::NetworkChecks do
+  RuntimeChecks::NetworkChecks.stubs(:hostname).returns("crowbar")
+  RuntimeChecks::NetworkChecks.stubs(:domain).returns("cloud.com")
+  RuntimeChecks::NetworkChecks.stubs(:fqdn).returns("crowbar.cloud.com")
+  RuntimeChecks::NetworkChecks.stubs(:ipv4_addrs).returns(Array.new(["192.168.52.10"]))
+  RuntimeChecks::NetworkChecks.stubs(:ipv6_addrs).returns(Array.new())
+
+  describe "#fqdn_detected?" do
+    it "returns true if fqnd is detectable" do
+      networkchecks = RuntimeChecks::NetworkChecks.new
+
+      expect(networkchecks.fqdn_detected?).to be true
+    end
+
+    it "returns false if the hostname is nil" do
+      RuntimeChecks::NetworkChecks.stubs(:hostname).returns(nil)
+      networkchecks = RuntimeChecks::NetworkChecks.new
+
+      expect(networkchecks.fqdn_detected?).to be false
+    end
+
+    it "returns false if the domain is nil" do
+      RuntimeChecks::NetworkChecks.stubs(:domain).returns(nil)
+      networkchecks = RuntimeChecks::NetworkChecks.new
+
+      expect(networkchecks.fqdn_detected?).to be false
+    end
+  end
+
+  describe "#ip_address_resolved?" do
+    it "returns true if the ip address is resolvable" do
+      networkchecks = RuntimeChecks::NetworkChecks.new
+
+      expect(networkchecks.ip_address_resolved?).to be true
+    end
+
+    it "retuns false if no ip address is resolvable" do
+      RuntimeChecks::NetworkChecks.stubs(:ipv4_addrs).returns(Array.new())
+      RuntimeChecks::NetworkChecks.stubs(:ipv6_addrs).returns(Array.new())
```

Comment:
Style/MethodCallParentheses: Do not use parentheses for method calls with no arguments.